x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -1168,7 +1166,7 @@ private SearchRequest buildQueryToUpdateDestinationIndex() {
         }
 
         sourceBuilder.query(queryBuilder);
-        logger.debug(() -> new ParameterizedMessage("[{}] Querying for data: {}", getJobId(), sourceBuilder));
+        logger.debug("[{}] Querying {} for data: {}", getJobId(), request.indices(), sourceBuilder);
```

Comment:
I always thought that the logger would serialize the strings unless we provided a message provider. 

Consequently, if we weren't on debug level, we are serializing the sourceBuilder unnecessarily.